connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -24,97 +30,217 @@ import type { MicrosoftNodeModel } from "@connectors/lib/models/microsoft";
+
+  const rootResources =
+    await MicrosoftRootResource.listRootsByConnectorId(connectorId);
+
+  // get root folders and drives and drill down site-root and sites to their
+  // child drives, as MicrosoftNodeData
+  const rootFolderAndDriveNodes: MicrosoftNodeData[] = rootResources.filter(
+    (resource) =>
+      resource.nodeType === "folder" || resource.nodeType === "drive"
+  );
+
+  const rootSiteNodes: MicrosoftNodeData[] = rootResources.filter(
+    (resource) => resource.nodeType === "site"
+  );
+
+  if (rootResources.some((resource) => resource.nodeType === "sites-root")) {
+    const msSites = await getSites(client);
+    rootSiteNodes.push(
+      ...msSites.map(
+        (site): MicrosoftNodeData => ({
+          itemApiPath: getSiteApiPath(site),
+          nodeType: "site",
+        })
+      )
+    );
+  }
+
+  const siteDriveNodes: MicrosoftNodeData[] = (
+    await concurrentExecutor(
+      rootSiteNodes,
+      async (site) => {
+        const msDrives = await getDrives(
+          client,
+          microsoftInternalIdFromNodeData(site)
+        );
+        return msDrives.map((drive) => ({
+          itemApiPath: getDriveApiPath(drive),
+          nodeType: "drive" as const,
+        }));
+      },
+      { concurrency: 5 }
+    )
+  ).flat();
+
+  // remove duplicates
+  const allNodes = [...siteDriveNodes, ...rootFolderAndDriveNodes].reduce(
+    (acc, current) => {
+      const x = acc.find(
+        (item) =>
+          item.nodeType === current.nodeType &&
+          item.itemApiPath === current.itemApiPath
+      );
+      if (!x) {
+        return acc.concat([current]);
+      } else {
+        return acc;
+      }
+    },
+    [] as MicrosoftNodeData[]
+  );
+
+  return MicrosoftNodeResource.batchGetOrCreateFromNodesData(
+    connectorId,
+    allNodes
+  );
+}
+
+export async function markNodeAsVisited(
+  connectorId: ModelId,
+  node: MicrosoftNodeType
+) {
+  const connector = await ConnectorResource.fetchById(connectorId);
+  if (!connector) {
+    throw new Error(`Connector ${connectorId} not found`);
+  }
+
+  await MicrosoftNodeResource.upsert({
+    ...node,
+    lastSeenTs: new Date(),
+  });
+}
+
+const FILES_SYNC_CONCURRENCY = 10;
+
+/**
+ * Given a drive or folder, sync files under it and returns a list of folders to sync
+ */
+export async function syncFiles({
   connectorId,
-  dataSourceConfig,
+  parent,
+  startSyncTs,
 }: {
   connectorId: ModelId;
-  dataSourceConfig: DataSourceConfig;
-}): Promise<void> {
+  parent: MicrosoftNodeType;
+  startSyncTs: number;
+}) {
   const connector = await ConnectorResource.fetchById(connectorId);
-
   if (!connector) {
-    throw new Error(`Connector with id ${connectorId} not found`);
+    throw new Error(`Connector ${connectorId} not found`);
   }
 
-  const config =
+  if (parent.nodeType !== "folder" && parent.nodeType !== "drive") {
+    throw new Error(`Parent node is not a folder or drive: ${parent.nodeType}`);
+  }
+
+  const providerConfig =
```

Comment:
he did, but as a private field
@spolu is it fine to make the field non private so that can be grabbed here? or was there a specific intent behind making it private that should leave it private? (vs e.g. readonly?)

Reply:
Not sure I understand how it's related to configuration here?

Reply:
But yeah the idea is that the connector always has its configuration loaded. So if we need it we should not attempt a refetch for sure

Reply:
On the line below (L. 173) we explicitly fetch the `MicrosoftConfigurationResource`, where as it's already done in the `ConnectorResource`, can we use it then?

https://github.com/dust-tt/dust/blob/8d5043d753502f075db664f3a0fd077d3ef5c1bd/connectors/src/resources/connector/microsoft.ts#L46-L52

Reply:
The config contains pdfEnabled, etc. I'll switch the field to readonly

Reply:
Yeah R/O is fine 👍

Reply:
actually can't do RO since it's set by a postFetch hook
wrote a getter

Reply:
actually actually
the getter doesn't work as-is, it'd be a larger refactor
will take care of it in a follow-up

Reply:
could not find an easy way to type, so went with an `as` (PR https://github.com/dust-tt/dust/pull/6151)